kotest-runner/kotest-runner-junit5/src/jvmMain/kotlin/io/kotest/runner/junit/platform/gradle/GradleClassMethodRegexTestFilter.kt

```diff
@@ -18,41 +18,58 @@ class GradleClassMethodRegexTestFilter(private val patterns: List<String>) : Tes
       }
    }
 
-   fun match(pattern: String, descriptor: Descriptor): Boolean {
-      val (pck, classname, path) = GradleTestPattern.parse(pattern)
+   private fun match(pattern: String, descriptor: Descriptor): Boolean {
+      val (prefixWildcard, pck, classname, path) = GradleTestPattern.parse(pattern)
       return when (descriptor) {
-         is Descriptor.TestDescriptor -> descriptor.parts().take(path.size) == path
+         is Descriptor.TestDescriptor -> when (path) {
+           null -> false
+           else -> descriptor.path(false).value.startsWith(path)
+         }
+
          is Descriptor.SpecDescriptor -> when {
+            pck != null && classname != null && prefixWildcard -> descriptor.kclass.qualifiedName?.contains("$pck.$classname") ?: false
             pck != null && classname != null -> descriptor.kclass.qualifiedName == "$pck.$classname"
+            pck != null && prefixWildcard -> descriptor.kclass.qualifiedName?.contains(pck) ?: true
             pck != null -> descriptor.kclass.qualifiedName?.startsWith(pck) ?: true
+            classname != null && prefixWildcard -> descriptor.kclass.simpleName?.contains(classname) ?: false
             classname != null -> descriptor.kclass.simpleName == classname
             else -> true
          }
       }
    }
 }
 
-data class GradleTestPattern(val pckage: String?, val classname: String?, val path: List<String>) {
+data class GradleTestPattern(
+   val suffixWildcard: Boolean,
+   val pckage: String?,
+   val classname: String?,
+   val path: String?,
+) {
    companion object {
 
       // if the regex starts with a lower case character, then we assume it is in the format package.Class.testpath
       // otherwise, we assume it is in the format Class.testpath
       // the .testpath is always optional, and at least Class or package must be specified
+      // additionally, patterns can start with a * in which case the pattern matches suffixes
       fun parse(pattern: String): GradleTestPattern {
+         require(pattern.isNotBlank())
+
+         val prefixWildcard = pattern.startsWith("*") || pattern.startsWith("*.")
```

Comment:
If it starts with *. it definitely starts with *, right?
```suggestion
         val prefixWildcard = pattern.startsWith("*")
```

Reply:
Yes but  I need rid of the . too so we don't end up with the package name being .something

Reply:
What I mean is
If a pattern start with `*.`, it will always start with `*`.
The second statement won't ever be true if the first one isn't true

Reply:
Ah yes :)